segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AbstractConnectionListener.java

```diff
@@ -224,4 +272,35 @@ public void close() {
             tlsCertFileModificationMonitor.stopMonitoring();
         }
     }
+
+    // A contributor for managing health of a connection listener.
+    private static class ConnectionListenerHealthContributor extends AbstractHealthContributor {
+        private final AbstractConnectionListener listener;
+        private final String host;
+        private final int port;
+
+        private ConnectionListenerHealthContributor(String connectionType, AbstractConnectionListener listener, String host, int port) {
+            super(connectionType);
```

Comment:
If you leave only the `listener` parameter for the constructor, you can just write:
```
super(listener.getClass().getSimpleName());
this.listener = listener;
```
That's it, you can use the host and port with `listerner.host` and `listener.port`, and you can get rid of these fields. Also, I think that you can add the `@NonNull` annotation to the `listener` parameter in the constructor, as you do not want to have a `null` listener.

Reply:
This definitely makes it simpler, thank you.